imap_processing/hi/l2/hi_l2.py

```diff
@@ -0,0 +1,192 @@
+"""IMAP-HI L2 processing module."""
+
+import logging
+from pathlib import Path
+from typing import Literal
+
+import numpy as np
+import xarray as xr
+
+from imap_processing.ena_maps.ena_maps import HiPointingSet, RectangularSkyMap
+from imap_processing.spice.geometry import SpiceFrame
+
+logger = logging.getLogger(__name__)
+
+
+def generate_hi_map(
+    psets: list[str | Path],
+    geometric_factors_path: str | Path,
+    esa_energies_path: str | Path,
+    cg_corrected: bool = False,
+    direction: Literal["ram", "anti-ram", "full"] = "full",
+    map_spacing_deg: int = 4,
+) -> xr.Dataset:
+    """
+    High level IMAP-Hi L2 processing function.
+
+    Parameters
+    ----------
+    psets : list of str or Path
+        List of input PSETs to make a map from.
+    geometric_factors_path : str or Path
+        Where to get the geometric factors from.
+    esa_energies_path : str or Path
+        Where to get the energies from.
+    cg_corrected : bool, optional
+        Whether to apply Compton-Getting correction to the energies. Defaults to
+        False.
+    direction : str, optional
+        Apply filtering to PSET data include ram or anti-ram or full spin data.
+        Defaults to "full".
+    map_spacing_deg : int, optional
+        Pixel spacing of the output map in degrees. Defaults to 4.
+
+    Returns
+    -------
+    l2_dataset : xr.Dataset
+        Level 2 IMAP-Hi dataset ready to be written to a CDF file.
+    """
+    rect_map = RectangularSkyMap(
+        spacing_deg=map_spacing_deg, spice_frame=SpiceFrame.ECLIPJ2000
+    )
+
+    # TODO: Implement Compton-Getting correction
+    if cg_corrected:
+        raise NotImplementedError
+    # TODO: Implement directional filtering
+    if direction != "full":
+        raise NotImplementedError
```

Comment:
Nice hooks! 

This isn't necessary until you start implementing it, but for direction/spin_phase, maybe have an enum?